cli/app/app.go

```diff
@@ -39,130 +40,133 @@ import (
 
 const (
 	DefaultComposeFile = "docker-compose.yml"
+	DefaultProvider    = "kubernetes"
 )
 
 var inputFormat = "compose"
 
-// Hook for erroring and exit out on warning
-type errorOnWarningHook struct{}
+func validateFlags(c *cli.Context, opt *kobject.ConvertOptions) {
 
-func (errorOnWarningHook) Levels() []logrus.Level {
-	return []logrus.Level{logrus.WarnLevel}
-}
-
-func (errorOnWarningHook) Fire(entry *logrus.Entry) error {
-	logrus.Fatalln(entry.Message)
-	return nil
-}
-
-// BeforeApp is an action that is executed before any cli command.
-func BeforeApp(c *cli.Context) error {
-
-	if c.GlobalBool("verbose") {
-		logrus.SetLevel(logrus.DebugLevel)
-	} else if c.GlobalBool("suppress-warnings") {
-		logrus.SetLevel(logrus.ErrorLevel)
-	} else if c.GlobalBool("error-on-warning") {
-		hook := errorOnWarningHook{}
-		logrus.AddHook(hook)
+	if opt.OutFile == "-" {
+		opt.ToStdout = true
+		opt.OutFile = ""
 	}
-	return nil
-}
 
-func validateFlags(opt kobject.ConvertOptions, singleOutput bool, dabFile, inputFile string) {
 	if len(opt.OutFile) != 0 && opt.ToStdout {
 		logrus.Fatalf("Error: --out and --stdout can't be set at the same time")
 	}
+
 	if opt.CreateChart && opt.ToStdout {
 		logrus.Fatalf("Error: chart cannot be generated when --stdout is specified")
 	}
+
 	if opt.Replicas < 0 {
 		logrus.Fatalf("Error: --replicas cannot be negative")
 	}
-	if singleOutput {
-		count := 0
+
+	dabFile := c.GlobalString("bundle")
+
+	if len(dabFile) > 0 {
+		inputFormat = "bundle"
+		opt.InputFile = dabFile
+	}
+
+	if len(dabFile) > 0 && len(opt.InputFile) > 0 && opt.InputFile != DefaultComposeFile {
+		logrus.Fatalf("Error: 'compose' file and 'dab' file cannot be specified at the same time")
+	}
+}
+
+func validateControllers(opt *kobject.ConvertOptions) {
+
+	singleOutput := len(opt.OutFile) != 0 || opt.OutFile == "-" || opt.ToStdout
+
+	if opt.Provider == "kubernetes" {
+		// create deployment by default if no controller has been set
+		if !opt.CreateD && !opt.CreateDS && !opt.CreateRC {
+			opt.CreateD = true
+		}
+		if opt.CreateDeploymentConfig {
```

Comment:
Oh yes, now this is not needed since command line handling will take care of it :)